### PR TITLE
Custom Item Limb Framework + Gracia's Aut'akh Arm

### PR DESCRIPTION
**Pull request by Geevies**
* You can now view your custom items from within the loadout.

Also implements the framework necessary for custom items to be limbs, and to automatically replace the limb it's meant to replace.

Implements the Aut'akh arm mentioned in this request: https://forums.aurorastation.org/topic/21092-gracias-autakh-arm

### CODE REVIEW

**Comment by Geevies** [comment]
!review

**Comment by Geevies** [comment]
marking as DNM. i want the code reviewed but the arm itself should not be merged until the custom item is accepted. once the decision is made i'll remove the tag. in the event the custom item arm is denied, i'll remove it arm itself from the PR, i mainly used it to test the framework in the first place

File: code/modules/organs/organ_external.dm [hunk 1]
```diff
@@ -1242,9 +1254,10 @@ Note that amputating the affected organ does in fact remove the infection from t
 	limb_flags &= ~ORGAN_CAN_BREAK
 	get_icon()
 	unmutate()
-	for (var/obj/item/organ/external/T in children)
-		if(T)
-			T.robotize(company)
+	if(robotize_children)
+		for (var/obj/item/organ/external/T in children)
+			if(T)
```
Comment (FluffyGhoster) [inline code/modules/organs/organ_external.dm:1259]: This checks makes no sense? It should already not be null from the for loop

File: code/modules/customitems/item_spawning.dm [hunk 2]
```diff
@@ -213,68 +213,102 @@
 
 	return item
 
+GLOBAL_LIST_EMPTY(character_id_to_custom_items_mapping)
+
 //gets the relevant list for the key from the listlist if it exists, check to make sure they are meant to have it and then calls the giving function
-/proc/equip_custom_items(var/mob/living/carbon/human/M)
+/proc/equip_custom_items(var/mob/living/carbon/human/target_mob, var/character_id, var/player_ckey)
```
Comment (FluffyGhoster) [inline code/modules/customitems/item_spawning.dm:219]: DMDoc

File: code/modules/customitems/item_spawning.dm [hunk 3]
```diff
@@ -213,68 +213,102 @@
...
+				ci.item_path = text2path(char_item_query.item[5])
+				ci.item_data = json_decode(char_item_query.item[6]) //TODO: try/catch
+				ci.req_titles = json_decode(char_item_query.item[7]) //TODO: try/catch
+				ci.additional_data = char_item_query.item[8]
+
+				custom_items_list += ci
+
+			GLOB.character_id_to_custom_items_mapping[character_id] = custom_items_list
+
+		var/list/cached_custom_items = GLOB.character_id_to_custom_items_mapping[character_id]
```
Comment (FluffyGhoster) [inline code/modules/customitems/item_spawning.dm:253]: Useless var

**Review by DreamySkrell — APPROVED** [review]

**Review by FluffyGhoster — APPROVED** [review]
In the light of the Omnissiah, I fear nothing

**Pull request merged**